pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1156,7 +1156,7 @@ func applyUpstreamTLSSettings(env *model.Environment, cluster *apiv2.Cluster, tl
 	}
 
 	// convert to transport socket matcher if the mode was auto detected
-	if tls.Mode == networking.TLSSettings_ISTIO_MUTUAL && mtlsCtxType == autoDetected {
+	if tls.Mode == networking.TLSSettings_ISTIO_MUTUAL && mtlsCtxType == autoDetected && util.IsIstioVersionGE14(proxy) {
```

Comment:
unless you plan to CP this into 1.3 release branch, I don't think this check is needed. right? If you only CP into 1.4

Reply:
Pilot 1.4 must work with Envoy 1.3

Reply:
Oh I see, thanks for clarifying, I missed that...